Makefile

```diff
@@ -39,12 +39,21 @@ endif
 
 all: all-libs all-tests
 
-all-libs:
+all-libs: check-fmt
 	$(CARGO) build $(BUILD) $(CARGO_FEATURES) $(CARGO_FLAGS)
 
-all-tests:
+all-tests: check-fmt
 	$(CARGO) build  --tests $(BUILD) $(CARGO_FEATURES) $(CARGO_FLAGS)
 
+check-fmt: check-fmt-c check-fmt-rust
+
+check-fmt-c:
+	$(shell find include/ -name "*.h" -type f -print0 | xargs -0 clang-format --fallback-style=Microsoft --dry-run -Werror )
```

Comment:
Just .h files?

Reply:
We just have header files in the project right now, but I'm happy to add ".c" files to the rule.

Reply:
Yes, I think we should, so they get checked when any are added.  Maybe we should do C++ files as well?

Reply:
Sure. I'll add `.c`, `.cpp` and `.h` files. Am I missing any file extension?

Anyways, I'll apply these changes once we agree on the formatting style that we are going to use.